packages/@snowpack/plugin-babel/plugin.js

```diff
@@ -1,11 +1,27 @@
-const babel = require("@babel/core");
+const fs = require('fs');
+const babel = require('@babel/core');
+
+module.exports = function plugin(_, options) {
+  // options validation
+  if (options) {
+    if (typeof options !== 'object') throw new Error(`options isn’t an object. Please see README.`);
+    if (options.input && !Array.isArray(options.input))
+      throw new Error(
+        `options.input must be an array (e.g. ['.js', '.mjs', '.jsx', '.ts', '.tsx'])`,
+      );
+    if (options.input && !options.input.length)
+      throw new Error(`options.input must specify at least one filetype`);
+  }
 
-module.exports = function plugin(config, options) {
   return {
-    defaultBuildScript: "build:js,jsx,ts,tsx",
-    async build({ contents, filePath, fileContents }) {
-      const result = await babel.transformAsync(contents || fileContents, {
-        filename: filePath,
+    name: '@snowpack/plugin-babel',
+    resolve: {
+      input: options.input || ['.js', '.mjs', '.jsx', '.ts', '.tsx'],
+      output: ['.js'], // always export JS
+    },
+    async load({filePath}) {
+      if (!filePath) return;
+      const result = await babel.transformFileAsync(filePath, {
```

Comment:
minor change: swapped this to `transformFileAsync`, which skips us having to read the file ourselves. I don’t think it’s faster but it saves a line 🤷🏻

Reply:
you never know!